code/game/objects/items/stacks/sheets/mineral.dm

```diff
@@ -218,10 +218,21 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
 
 /obj/item/stack/sheet/mineral/plasma/welder_act(mob/user, obj/item/I)
 	if(I.use_tool(src, user, volume = I.tool_volume))
-		message_admins("Plasma sheets ignited by [key_name_admin(user)]([ADMIN_QUE(user,"?")]) ([ADMIN_FLW(user,"FLW")]) in ([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)",0,1)
-		log_game("Plasma sheets ignited by [key_name(user)] in ([x],[y],[z])")
-		investigate_log("was <font color='red'><b>ignited</b></font> by [key_name(user)]","atmos")
-		fire_act()
+		log_and_set_aflame(user, I)
+	return TRUE
+
+/obj/item/stack/sheet/mineral/plasma/attackby(obj/item/I, mob/living/user, params)
+	if(is_hot(I))
+		log_and_set_aflame(user, I)
+	else
+		return ..()
+
+/obj/item/stack/sheet/mineral/plasma/proc/log_and_set_aflame(mob/user, obj/item/I)
+	message_admins("Plasma sheets ignited by [key_name_admin(user)]([ADMIN_QUE(user, "?")]) ([ADMIN_FLW(user, "FLW")]) in ([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)")
+	log_game("Plasma sheets ignited by [key_name(user)] in coord(SRC)")
```

Comment:
dm is a case sensitive language. coord != COORD. SRC != src. coord is not a defined proc or define.
It also needs to be put inside `[]`. Anything inside `[]`'s are considered code in strings. So you can do this:
`"answer to 1 + 2 = [1 + 2]"` which will become `"answer to 1 + 2 = 3"`

Reply:
Updated to fix it, thanks for the explanation